authors/apps/authentication/views.py

```diff
@@ -135,56 +146,154 @@ def update(self, request, *args, **kwargs):
+
+class PasswordResetView(APIView):
     """
-    Activate an account
+      An Api View which provides a method to verifiy that a given pw-reset token is valid before actually confirming the
+      reset.
     """
-    def get(self, request, **kwargs):
 
-        try:
-            uid = force_text(urlsafe_base64_decode(kwargs['uidb64']))
-            user = User.objects.get(pk=uid)
+    serializer_class = PasswordSerializer
 
-        except User.DoesNotExist:
-            return Response("Activation failed")
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        token = kwargs["token"]
+        password = serializer.validated_data["password"]
 
-        if user and default_token_generator.check_token(user, kwargs['token']):
-            user.is_email_verified = True
-            user.is_active = True
-            user.save()
-            return Response("Activation successful")   
-        else:
-            return Response("Activation link has expired")         
+        # get token validation time
+        password_reset_token_validation_time = (
+            get_password_reset_token_expiry_time()
+        )
+
+        # find token
+        reset_password_token = ResetPasswordToken.objects.filter(
+            key=token
+        ).first()
+
+        if reset_password_token is None:
+            return Response(
+                {"error": "invalid token"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        # check expiry date
+        expiry_date = reset_password_token.created_at + timedelta(
+            hours=password_reset_token_validation_time
+        )
+
+        if timezone.now() > expiry_date:
+            # delete expired token
+            reset_password_token.delete()
+            return Response(
+                {"error": "Token expired"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        reset_password_token.user.set_password(password)
+        reset_password_token.user.save()
+
+        return Response(
+            {"message": "Password reset succesfully"}, status.HTTP_200_OK
+        )
+
+
+@receiver(reset_password_token_created)
+def password_reset_token_created(
+    sender, instance, reset_password_token, *args, **kwargs
+):
+    """
+    Handles password reset tokens
+    When a token is created, an e-mail needs to be sent to the user
+    sender: View Class that sent the signal
+    instance: View Instance that sent the signal
+    reset_password_token: Token Model Object    
+    """
+
+    text_content = "Password Reset"
+    template_name = "user_reset_password.html"
+
+    context = {
+        "current_user": reset_password_token.user,
+        "username": reset_password_token.user.username,
+        "email": reset_password_token.user.email,
+        "reset_password_url": "{}{}{}".format(
+            settings.WEB_HOST,
+            reverse("password_reset:reset-password-request"),
```

Comment:
@bisonlou. I received the email for password reset when I consumed the POST `https://ah-backend-dojo-dev.herokuapp.com/api/password_reset/reset_password/` with my email in the request body, however the redirection link I received in the email is to `https://ah-backend-dojo-dev.herokuapp.com/api/password_reset/5e6295321de2aece9d102` instead of https://ah-backend-dojo-dev.herokuapp.com/api/reset-password/5e6295321de2aece9d102 I think the problem might be with the reverse url.

Reply:
It is fixed now. Check again and review